packages/db/src/models/base.ts

```diff
@@ -256,9 +270,21 @@ class BasePgModel<ItemType, RecordType extends BaseRecord> {
     updateParams: Partial<RecordType>,
     returning: Array<string> = []
   ) {
-    return await knexOrTransaction(this.tableName)
+    const records = await knexOrTransaction(this.tableName)
       .where(whereClause)
-      .update(updateParams, returning);
+      .update(updateParams, returning) as RecordType[];
+
+    if (typeof records !== 'number') {
```

Comment:
Actually, no (thanks knex):

https://knexjs.org/#Builder-returning

would imply that, however *if* returning uses the default `[]` knex appears to return the number of rows updated based on a quick bit of unit test proofing/debugging.   

Neat, eh?   

Good catch on the typing issue, the implied unreachable code feels like something that should have been caught by lint/compilation.    I'll see if I can sort that, but it might be an error on knex typing, or ..... ????